.github/workflows/build.yml

```diff
@@ -95,6 +95,46 @@ jobs:
         shell: bash
         run: |
           pytest -sv tests/unit
+      - name: Check for Secret availability
+        id: secret-check
+        if: ${{ contains(fromJSON('["3.9"]'), matrix.python) }}
+        # perform secret check & put boolean result as an output
+        shell: bash
+        run: |
+          if [ -z "${{ secrets.encrypted_d17283647768_key }}" ]  || [ -z "${{ secrets.encrypted_d17283647768_iv }}" ]; then
+            echo "secrets_available=false" >> $GITHUB_OUTPUT;
+          else
+            echo "secrets_available=true" >> $GITHUB_OUTPUT;
+          fi
+      - name: Install AWS Tools
+        id: aws-check
+        if: ${{ steps.secret-check.outputs.secrets_available == 'true' && !(startsWith(matrix.os, 'windows')) }}
+        shell: bash
+        run: |
+          echo "run_opentelemetry=true" >> $GITHUB_OUTPUT;
+
+          # AWS CLI is pre-installed on github hosted runners
+          # curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
+          # unzip awscliv2.zip
+          # sudo ./aws/install
+          # curl "https://s3.amazonaws.com/session-manager-downloads/plugin/latest/ubuntu_64bit/session-manager-plugin.deb" -o "session-manager-plugin.deb"
+          # sudo dpkg -i session-manager-plugin.deb
+      - name: Decrypt AWS Config
+        if: ${{ steps.aws-check.outputs.run_opentelemetry == 'true' }}
+        shell: bash
+        run: |
+          openssl aes-256-cbc -K ${{ secrets.encrypted_d17283647768_key }} -iv ${{ secrets.encrypted_d17283647768_iv }} -in test.awsConfig.enc -out test.awsConfig -d
```

Comment:
The file contains the synapse PAT.

I chose to encrypt over GH action secrets because I'm not an admin on this project and don't have access to add a new secret - Though I'm not opposed to making this change

Reply:
Thanks both - that is a great idea.  We would need to add in the other parameters that the synapse config file supports within github secrets.  That would definitely be cleaner than encoding / decrypting when we need to make changes, but it may not simplify the github actions due to all the other parameters.

@BryanFauble I'll get you added to the admin list.

Reply:
if you are using a synapse PAT then i assume that means you would need to create a synapse service user for every repo or possibly every test?

Reply:
We created a team-wide synapse service user to serve this purpose. Since service catalog instances cannot be shared we needed a single entry point for all team members and this automation.

The only purpose of this service user is to allow connecting to a service catalog instance. There is a separate `test` service account on the dev synapse repo that is responsible for executing the requests with the REST API.

Reply:
@thomasyu888 I created https://sagebionetworks.jira.com/browse/SYNPY-1327 to track migrating the `test.synapseConfig` file over.

For this awsConfig file I moved over to creating it from GH secrets: https://github.com/Sage-Bionetworks/synapsePythonClient/pull/1009/commits/af9ee486546a620f355dfc036e18dcf12f6ea4a6